tests/integration_tests/sqllab_tests.py

```diff
@@ -568,7 +568,9 @@ def test_sql_limit(self):
             query_limit=test_limit + 1,
         )
         self.assertEqual(len(data["data"]), test_limit)
-        self.assertEqual(data["query"]["limitingFactor"], LimitingFactor.QUERY)
+        self.assertEqual(
+            data["query"]["limitingFactor"], LimitingFactor.QUERY_AND_DROPDOWN
```

Comment:
This is now wrong... if the query is `SELECT * FROM birth_names LIMIT 1` and `query_limit` is 2, then the limiting factor should be `LimitingFactor.QUERY`, as before.

You probable need to adjust (or remove) the logic where we parse the query to extract the limit, since it's now being done in `SqlJsonExecutionContext`.

Reply:
Yeah I wanted to talk to you about this. Should I transfer the limiting factor logic into `SqlJsonExecutionContext`?

Reply:
I think so, I need to look at the code and remember how this all works. :-P